### PR TITLE
feat(app): add GEN2 images to change pipette

### LINKED ISSUE #3630: Pipette Gen2: View Gen2 Image When Changing Pipettes
As a user trying to change pipettes, I would like to see an accurate image reflecting my Gen2 pipette when following instructions to attach or detach my pipette.

## Acceptance Criteria
- [ ] Replace 1st generation images with Gen2 images for attach/detach, single/multi, and left/right images.

## Design
- Will need new assets

### PR DESCRIPTION
**Pull request by b-cooper**
## Overview

Add images for GEN2 single and multi pipettes when changing or viewing attached pipettes.

Closes #3630

## Changelog

- Update `InstrumentDiagram` in CL to allow for image differentiation on 3 factors (mount, displayCategory, channels)
- Update `InstructionStep` in RA to allow for image differentiation on 5 factors (mount, displayCategory, channels, direction, diagram)
- Update all usages of the above components and their dependents to pass down the required data
- Add first call to `require()` passing a template string as a param, add flow config to ignore this case
- Add new image assets for GEN2 pipette thumbs and change pipette diagrams

### CODE REVIEW

File: app/src/components/ChangePipette/styles.css [hunk 1]
```diff
@@ -143,3 +142,8 @@
   padding-left: 2.5rem;
   padding-right: 2.5rem;
 }
+
+.flipped_image {
+  transform: scaleX(-1);
+  filter: FlipH;
```
Comment (mcous) [inline app/src/components/ChangePipette/styles.css]: Looking at [caniuse](https://caniuse.com/#feat=transforms2d) I think `transform: scaleX(-1)` should be enough to get us coverage all the way back to IE9 (with our autoprefixing that's already in place). We should get together and propose an actual browser support spec for our projects

File: components/src/instrument-diagram/InstrumentDiagram.js [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+import type { Mount } from '../robot-types'
 import singleSrc from './pipetteSingle.png'
 import multiSrc from './pipetteMulti.png'
+import singleGEN2Src from './pipetteGEN2Single.png'
+import multiGEN2Src from './pipetteGEN2Multi.png'
+import styles from './instrument.css'
 
 type Props = {
-  channels?: number,
+  pipetteSpecs?: ?PipetteNameSpecs | ?PipetteModelSpecs,
```
Comment (mcous) [inline components/src/instrument-diagram/InstrumentDiagram.js]: I think in this case we'll be good, but we should consider making `PipetteNameSpecs` and `PipetteModelSpecs` exact objects to make sure unions behave a little better

File: components/src/instrument-diagram/InstrumentDiagram.js [hunk 3]
```diff
@@ -1,21 +1,45 @@
...
+  const { pipetteSpecs, mount } = props
+  const { displayCategory, channels } = pipetteSpecs || {}
 
+  let imgSrc
+  switch (displayCategory) {
+    case 'GEN2': {
+      imgSrc = channels === 1 ? singleGEN2Src : multiGEN2Src
+      break
+    }
+    default:
```
Comment (mcous) [inline components/src/instrument-diagram/InstrumentDiagram.js]: What's the purpose of the `default` above `case 'OG'` here? Some sort of flow exhaustiveness check?

Comment (b-cooper) [reply components/src/instrument-diagram/InstrumentDiagram.js]: Yup, I was reasoning that it should fall back to OG images, but I can see an argument for displaying nothing if the displayCategory is missing

File: app/src/components/ChangePipette/InstructionStep.js [hunk 4]
```diff
@@ -15,85 +18,36 @@ type DiagramProps = {
...
   step: 'one' | 'two',
   children: React.Node,
 }
 
-type Channels = 'single' | 'multi'
+export function getDiagramsSrc(props: DiagramProps) {
+  const { channels, displayCategory, direction, mount, diagram } = props
+  const channelsKey = channels === 8 ? 'multi' : 'single'
 
-// TODO(mc, 2018-04-06): there must be a better way...
```
[the remark this answers is not in the document]
Comment (Kadee80) [reply app/src/components/ChangePipette/InstructionStep.js]: 🤕

**Pull request merged**